SoftLayer/CLI/hardware/detail.py

```diff
@@ -107,6 +108,46 @@ def cli(env, identifier, passwords, price):
             pass_table.add_row([item['username'], item['password']])
         table.add_row(['remote users', pass_table])
 
+    if components:
+
+        _mask = 'id,hardwareComponentModel[name,' \
+                'hardwareGenericComponentModel[description,hardwareComponentType[keyName]],' \
+                'firmwares[createDate,version]]'
+
+        _filter = {"components": {
+            "hardwareComponentModel": {
+                "firmwares": {
+                    "createDate": {
+                        "operation": "orderBy",
+                        "options": [
+                            {
+                                "name": "sort",
+                                "value": [
+                                    "DESC"
+                                ]
+                            },
+                            {
+                                "name": "sortOrder",
+                                "value": [
+                                    1
+                                ]}]}
+                }}}}
+        components = hardware.get_components(identifier, _mask, _filter)
```

Comment:
```suggestion
        components = hardware.get_components(identifier)
```
after moving `_mask` and `_filter` within `get_components(identifier)`